### PR TITLE
Implement cancel method for IQMJob

### PR DESCRIPTION
**Pull request by jkotilahti**

### CODE REVIEW

**Comment by jkotilahti** [comment]
(CI will fail until https://github.com/iqm-finland/iqm-client/pull/89 is merged)

File: src/qiskit_iqm/iqm_job.py [hunk 1]
```diff
@@ -103,8 +111,19 @@ def submit(self):
...
-        raise NotImplementedError('Canceling jobs is currently not supported.')
+    def cancel(self) -> bool:
+        """Attempt to cancel the job.
+
+        Returns:
+            True if the job was cancelled successfully, False otherwise
+        """
+        try:
+            self._client.abort_job(uuid.UUID(self._job_id))
+            print(f'Job {self._job_id} cancelled successfully.')
```
Comment (q-mat-beu) [inline src/qiskit_iqm/iqm_job.py:122]: We should not use `print`. In the success case I would print nothing and in the failure case I would use the `warning` module.

Comment (jkotilahti) [reply src/qiskit_iqm/iqm_job.py]: Using `warnings.warn` instead of `print` now, also checking in unit tests that warning is produced when expected.

**Review by q-mat-beu — APPROVED** [review]

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,11 @@
 Changelog
 =========
 
+Version 11.0
```
Comment (q-mat-beu) [inline CHANGELOG.rst:5]: I think here it is actually not breaking because we have used `JobStatus` all along, so I would suggest a minor version update.

Comment (jkotilahti) [reply CHANGELOG.rst]: Updated version number.

**Review by q-mat-beu — APPROVED** [review]

**Pull request merged**